src/phylo/phylogeny.jl

```diff
@@ -0,0 +1,230 @@
+# phylo/phylogeny.jl
+# ==================
+#
+# Types and methods for phylogenetic trees.
+#
+# Part of the Bio.Phylo module.
+#
+# This file is a part of BioJulia. License is MIT: https://github.com/BioJulia/Bio.jl/blob/master/LICENSE.md
+
+
+
+# The Phylogeny Type
+# -------------------
+
+"""
+Phylogeny represents a phylogenetic tree.
+
+Fields:
+
+    - graph: A LightGraphs directional graph.
+
+    - ntaxa: A field to keep track of the number of tips in the phylogeny.
+
+    - rooted: A boolean field to keep track of whether the phylogeny is considered
+    rooted or not.
+
+    - rerootable: A boolean field to keep track of whether the phylogeny may be
+    rerooted or not.
+"""
+type Phylogeny{C, B}
+    graph::DiGraph
+    vertexindex::Indexer{Int}
+    vertexdata::Vector{C}
+    edgedata::Dict{Edge, B}
+    ntaxa::Int
+    rooted::Bool
+    rerootable::Bool
+end
+
+
+# Phylogeny Constructors
+# -----------------------
+
+"""
+    Phylogeny{C, B}(tips::Vector{Symbol}, ::Type{C}, ::Type{B})
+
+Construct a completely unresolved phylogeny, with no annotated data.
+
+The phylogeny will store annotation data of type C for tree nodes
+(clades and tips). It will also store data of type B associated with branches.
+
+# Examples
+```julia
+Phylogeny([:Human, :Chimp, :Dog, :Cat], Void, Void)
+Phylogeny([:Human, :Chimp, :Dog, :Cat], Float64, Float64)
+```
+"""
+function Phylogeny{C, B}(taxa::Vector{Symbol}, ::Type{C}, ::Type{B})
```

Comment:
I think type variables should come first in the arguments (`Phylogeny(::Type{C}, ::Type{B}, taxa::Vector{Symbol}`). The standard library seems to follow the convention.